### PR TITLE
SetIfEmpty: Fix set if empty non ascii chars

### PR DESCRIPTION
**Pull request by yaakovi**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/33445

## Description
improved handling of non-ASCII chars.
tested with playbook as the error is within the demisto class and cannot be tested via unit tests.

### CODE REVIEW

File: Packs/CommonScripts/Scripts/SetIfEmpty/SetIfEmpty.py [hunk 1]
```diff
@@ -10,7 +10,13 @@ def get_value_to_set(args):
     if value is None or (apply_if_empty and is_value_empty(value)):
         value = args.get('defaultValue')
 
-    return encode_string_results(value)
+    if isinstance(value, list):
+        for i, item in enumerate(value):
+            value[i] = encode_string_results(item)
```
Comment (reutshal) [inline Packs/CommonScripts/Scripts/SetIfEmpty/SetIfEmpty.py:15]:
```suggestion
        for key, item in enumerate(value):
            value[key] = encode_string_results(item)
```

File: Packs/CommonScripts/Scripts/SetIfEmpty/SetIfEmpty.py [hunk 2]
```diff
@@ -10,7 +10,13 @@ def get_value_to_set(args):
...
+
+        return value
+    else:
+        return encode_string_results(value)
```
Comment (reutshal) [inline Packs/CommonScripts/Scripts/SetIfEmpty/SetIfEmpty.py:19]: Can the value be a dict? will the encoding work on it?

**Review by reutshal — COMMENTED** [review]
Can you attach an image of the playbook?

File: Packs/CommonScripts/Scripts/SetIfEmpty/SetIfEmpty.py (hunk 2 above)
Comment (yaakovi) [reply Packs/CommonScripts/Scripts/SetIfEmpty/SetIfEmpty.py:19]: I was able to reproduce it only with a list of strings

File: Packs/CommonScripts/Scripts/SetIfEmpty/SetIfEmpty.py (hunk 1 above)
Comment (yaakovi) [reply Packs/CommonScripts/Scripts/SetIfEmpty/SetIfEmpty.py:15]: since `value` is a list, `i` stands for index.

**Comment by yaakovi** [comment]
> Can you attach an image of the playbook?

SetIfEmpty_-_non-ascii_chars_-_Test_Sat_Feb_06_2021

**Review by reutshal — APPROVED** [review]

**Pull request merged**